### ISSUE TITLE
ROCm 1.8 with tensorflow 1.3 2.5x slower than ROCm 1.7 on mnist

### ISSUE THREAD

**Issue by srondeauuwt** [RadeonOpenCompute/ROCm#416]
Using a Dell Precision 7910XL, a Vega 56 GPU, Ubuntu 16.04.4 LTS and 4.13.0-32-generic kernel, I tested convolution.py from the tensorflow models/tutorials/image/mnist/ folder, using python (vs. python3). With the 4.4 or 4.13 kernel and ROCm 1.7,  I got this per step:

Step 100 (epoch 0.12), 8.8 ms

Now with 1.8, I get this per step:

Step 100 (epoch 0.12), 22.0 ms

I was hoping to get better than 8.8 ms because our nvidia Titan XP gets about 4 ms.

Any ideas how I can get better performance or return to the 8.8 ms/step performance?

rocminfo:
``
=====================
HSA System Attributes
=====================
Runtime Version:         1.1
System Timestamp Freq.:  1000.000000MHz
Sig. Max Wait Duration:  18446744073709551615 (number of timestamp)
Machine Model:           LARGE
System Endianness:       LITTLE

==========
HSA Agents
==========
*******
Agent 1
*******
  Name:                    Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz
  Vendor Name:             CPU
  Feature:                 None specified
  Profile:                 FULL_PROFILE
  Float Round Mode:        NEAR
  Max Queue Number:        0
  Queue Min Size:          0
  Queue Max Size:          0
  Queue Type:              MULTI
  Node:                    0
  Device Type:             CPU
  Cache Info:
    L1:                      32768KB
  Chip ID:                 0
  Cacheline Size:          64
  Max Clock Frequency (MHz):3600
  BDFID:                   0
  Compute Unit:            16
  Features:                None
  Pool Info:
    Pool 1
      Segment:                 GLOBAL; FLAGS: KERNARG, FINE GRAINED
      Size:                    131952400KB
      Allocatable:             TRUE
      Alloc Granule:           4KB
      Alloc Alignment:         4KB
      Acessible by all:        TRUE
    Pool 2
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED
      Size:                    131952400KB
      Allocatable:             TRUE
      Alloc Granule:           4KB
      Alloc Alignment:         4KB
      Acessible by all:        TRUE
  ISA Info:
    N/A
*******
Agent 2
*******
  Name:                    gfx900
  Vendor Name:             AMD
  Feature:                 KERNEL_DISPATCH
  Profile:                 BASE_PROFILE
  Float Round Mode:        NEAR
  Max Queue Number:        128
  Queue Min Size:          4096
  Queue Max Size:          131072
  Queue Type:              MULTI
  Node:                    1
  Device Type:             GPU
  Cache Info:
    L1:                      16KB
  Chip ID:                 26751
  Cacheline Size:          64
  Max Clock Frequency (MHz):1590
  BDFID:                   1280
  Compute Unit:            56
  Features:                KERNEL_DISPATCH
  Fast F16 Operation:      FALSE
  Wavefront Size:          64
  Workgroup Max Size:      1024
  Workgroup Max Size Per Dimension:
    Dim[0]:                  67109888
    Dim[1]:                  83887104
    Dim[2]:                  0
  Grid Max Size:           4294967295
  Waves Per CU:            40
  Max Work-item Per CU:    2560
  Grid Max Size per Dimension:
    Dim[0]:                  4294967295
    Dim[1]:                  4294967295
    Dim[2]:                  4294967295
  Max number Of fbarriers Per Workgroup:32
  Pool Info:
    Pool 1
      Segment:                 GLOBAL; FLAGS: COARSE GRAINED
      Size:                    8372224KB
      Allocatable:             TRUE
      Alloc Granule:           4KB
      Alloc Alignment:         4KB
      Acessible by all:        FALSE
    Pool 2
      Segment:                 GROUP
      Size:                    64KB
      Allocatable:             FALSE
      Alloc Granule:           0KB
      Alloc Alignment:         0KB
      Acessible by all:        FALSE
  ISA Info:
    ISA 1
      Name:                    amdgcn-amd-amdhsa--gfx900
      Machine Models:          HSA_MACHINE_MODEL_LARGE
      Profiles:                HSA_PROFILE_BASE
      Default Rounding Mode:   NEAR
      Default Rounding Mode:   NEAR
      Fast f16:                TRUE
      Workgroup Max Dimension:
        Dim[0]:                  67109888
        Dim[1]:                  1024
        Dim[2]:                  16777217
      Workgroup Max Size:      1024
      Grid Max Dimension:
        x                        4294967295
        y                        4294967295
        z                        4294967295
      Grid Max Size:           4294967295
      FBarrier Max Size:       32
*** Done ***

**Comment by gstoner** [comment]
TensorFlow : 1.3 v - tf_cnn_benchmarks | ROCm 1.7.2 GA | Drop | ROCm 1.8 GA
-- | -- | -- | --
2500 , alexnet ; Batch size:  2048 global , 512 per device | 3196.48 | -1.61% | 3145.09
2500 , googlenet ; Batch size:  128 global 32 per device | 603.14 | 4.13% | 628.03
2500 , inception3 ; Batch size:  128 global 32 per device | 172.39 | -6.00% | 162.04
2500 , inception4 ; Batch size:  128 global 32 per device | 69.67 | -7.55% | 64.41
2550 , lenet5 ; Batch size:  128 global 32 per device | 8433.25 | -7.40% | 7809.51
2500 , overfeat ; Batch size:  128 global 32 per device | 226.33 | -0.07% | 226.17
resnet50 ; Batch size:  256 global 64 per device | 302.72 | -2.69% | 294.57
2500 , resnet152_v2 ; Batch size:  128 global 32 per device | 113.3 | 12.97% | 128
2500 , trivial ; Batch size:  128 global 32 per device | 6155.1 | -3.20% | 5957.96
2500 , vgg11 ; Batch size:  256 global  64 per device | 307.41 | -3.99% | 295.14

**Comment by sunway513** [comment]
Hi @srondeauuwt , could you try our latest TF1.11 release based on ROCm1.9?
Here's part of the log I can see using single MI25 GPU:
```Step 0 (epoch 0.00), 703.7 ms
Minibatch loss: 8.334, learning rate: 0.010000
Minibatch error: 85.9%
Validation error: 84.6%
Step 100 (epoch 0.12), 5.1 ms
Minibatch loss: 3.236, learning rate: 0.010000
Minibatch error: 6.2%
Validation error: 7.6%
Step 200 (epoch 0.23), 4.8 ms
Minibatch loss: 3.362, learning rate: 0.010000
Minibatch error: 10.9%
Validation error: 4.5%
Step 300 (epoch 0.35), 4.6 ms
Minibatch loss: 3.153, learning rate: 0.010000
Minibatch error: 3.1%
Validation error: 2.9%
Step 400 (epoch 0.47), 4.3 ms
Minibatch loss: 3.210, learning rate: 0.010000
Minibatch error: 6.2%
Validation error: 2.8%
Step 500 (epoch 0.58), 4.4 ms
Minibatch loss: 3.173, learning rate: 0.010000
Minibatch error: 6.2%
Validation error: 2.4%
Step 600 (epoch 0.70), 4.2 ms
Minibatch loss: 3.144, learning rate: 0.010000
Minibatch error: 3.1%
Validation error: 2.2%
Step 700 (epoch 0.81), 4.2 ms
Minibatch loss: 2.963, learning rate: 0.010000```

**Comment by srondeauuwt** [comment]
> Hi @srondeauuwt , could you try our latest TF1.11 release based on ROCm1.9?
> Here's part of the log I can see using single MI25 GPU:
> ```
> Minibatch loss: 8.334, learning rate: 0.010000
> Minibatch error: 85.9%
> Validation error: 84.6%
> Step 100 (epoch 0.12), 5.1 ms
> Minibatch loss: 3.236, learning rate: 0.010000
> Minibatch error: 6.2%
> Validation error: 7.6%
> Step 200 (epoch 0.23), 4.8 ms
> Minibatch loss: 3.362, learning rate: 0.010000
> Minibatch error: 10.9%
> Validation error: 4.5%
> Step 300 (epoch 0.35), 4.6 ms
> Minibatch loss: 3.153, learning rate: 0.010000
> Minibatch error: 3.1%
> Validation error: 2.9%
> Step 400 (epoch 0.47), 4.3 ms
> Minibatch loss: 3.210, learning rate: 0.010000
> Minibatch error: 6.2%
> Validation error: 2.8%
> ```

Finally got a chance to upgrade from Ubuntu 16.04 to 18.04, and from Tensorflow 1.3 to 1.11 and RocM 1.9.1. It is now running much better than when I complained about 1.8.1; e.g.:

Step 100 (epoch 0.12), 7.3 ms
Minibatch loss: 3.247, learning rate: 0.010000
Minibatch error: 6.2%
Validation error: 7.9%

but in the meantime, Nvidia's drivers improved for the Titan XP, and now the mark to beat is about 2.8 ms per 100 steps vs. the 4 ms/100 I was seeing before. I doubt if my Vega 56 is going to outpace or even keep pace with the XP. Soon I will be buying an RTX 2080 Ti, and it will be interesting to compare results with it.
